core/src/main/java/org/kiwix/kiwixmobile/core/downloader/ChunkUtils.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.downloader
+
+import org.kiwix.kiwixmobile.core.utils.StorageUtils.getFileNameFromUrl
+
+object ChunkUtils {
+  private const val ALPHABET = "abcdefghijklmnopqrstuvwxyz"
+  private const val ALPHABET_SIZE = 26
+  private const val ZIM_EXTENSION = ".zim"
+  const val PART = ".part.part"
+  const val CHUNK_SIZE = 1024L * 1024L * 1024L * 2L
+
+  fun getChunks(
+    url: String?,
+    contentLength: Long,
+    notificationID: Int
+  ): List<Chunk> {
+    val fileCount = getZimChunkFileCount(contentLength)
+    val filename = getFileNameFromUrl(url)
+    val fileNames = getZimChunkFileNames(filename, fileCount)
+    return generateChunks(contentLength, url, fileNames, notificationID)
+  }
+
+  private fun generateChunks(
+    contentLength: Long,
+    url: String?,
+    fileNames: Array<String?>,
+    notificationID: Int
+  ): List<Chunk> {
+    val ranges = fileNames.indices.map { it * (CHUNK_SIZE + 1) }.map { it..it + CHUNK_SIZE }
+    val filenamesAndRanges: List<Pair<String?, LongRange>> = fileNames.zip(ranges)
+    val chunks: MutableList<Chunk> = ArrayList()
+    filenamesAndRanges.mapTo(chunks, { (filename, currentRange) ->
+      val range: String = if (contentLength in currentRange) {
+        "${currentRange.first}-"
```

Comment:
hmmm so this is a maybe weird idea but we can move this conditional into the string itself
pseudocode
```
"$first-${if(blah) last else ""}"
```

and then it is probably short enough to inline

Reply:
I think that would be too much at the expense of understanding. In fact I could have used mapToindexed() or mapIndexed() function instead of creating currentRange ranges but I thought I would be difficult to comprehend. But if you insist, I can do that

Reply:
wait wait wait, what?

No, keep making the ranges, I have been insistent on that for a while.

that was pseudocode, I said `first` because I didn't want to type `currentRange.first`

Reply:
Sorry, Github is acting up. My previous comment was shown two times so I deleted it and now both are not there 😓 .

Here is it:
`I think that would be too much at the expense of understanding. In fact I could have used mapToindexed() or mapIndexed() function instead of creating currentRange ranges but I thought I would be difficult to comprehend. But if you insist, I can do that`

Reply:
And can you see my reply?

>wait wait wait, what?
No, keep making the ranges, I have been insistent on that for a while.
that was pseudocode, I said first because I didn't want to type currentRange.first